examples/lock-threads.yaml

```
@@ -0,0 +1,18 @@
+name: 'Lock Closed Threads'
+
+on:
+  schedule:
+    - cron: '42 5 * * 1' # 5:42 AM UTC every Monday
+  workflow_dispatch:
```

Let's also add repository dispatch. We have hooks for this method in shinycoreci. 

I didn't know about workflow_dispatch. We should prolly add workflow dispatch to any workflow with repository dispatch.

```suggestion
  repository_dispatch: { types: [lock_threads] }
  workflow_dispatch:
```